src/invidious/videos.cr

```diff
@@ -989,9 +989,15 @@ def fetch_video(id, region)
 
   # Try to pull streams from embed URL
   if info["reason"]?
-    embed_page = YT_POOL.client &.get("/embed/#{id}").body
-    sts = embed_page.match(/"sts"\s*:\s*(?<sts>\d+)/).try &.["sts"]? || ""
-    embed_info = HTTP::Params.parse(YT_POOL.client &.get("/get_video_info?html5=1&video_id=#{id}&eurl=https://youtube.googleapis.com/v/#{id}&gl=US&hl=en&sts=#{sts}").body)
+    required_parameters = URI::Params.new({
+      "video_id" => [id],
+      "eurl"     => ["https://youtube.googleapis.com/v/#{id}"],
+      "html5"    => ["1"],
+      "c"        => ["TVHTML5"],
+      "cver"     => ["6.20180913"],
```

Comment:
I don't think that putting each parameter into an array is required.

Reply:
Yeah that would be cleaner but unfortunately, the [`URI::Params.new`](https://crystal-lang.org/api/1.0.0/URI/Params.html#new(raw_params:Hash(String,Array(String)))-class-method) constructor requires a type of `Hash(String, Array(String)))`. I'll go ahead and switch to [`URI::Params.encode`](https://crystal-lang.org/api/1.0.0/URI/Params.html#encode(hash:Hash(String,String%7CArray(String)))-class-method) which does allow for just.